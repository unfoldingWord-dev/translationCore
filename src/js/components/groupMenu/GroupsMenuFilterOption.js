import React from 'react';
import PropTypes from 'prop-types';

class GroupsMenuFilterOption extends React.Component {

  handleCheckboxChange(event) {
    const value = event.target.checked;
    this.props.setFilter(this.props.name, value);
  }

  render() {
    const {
      name,
      text,
      icon,
      checked,
      disabled
    } = this.props;
    return (
      <label className={"option"+(disabled?" disabled":"")}>
        <span className="option-checkbox">
          <input type="checkbox" name={name} checked={checked} disabled={disabled} onChange={this.handleCheckboxChange.bind(this)} />
        </span>
        <span className="option-icon">
          {icon}
        </span>
        <span className="option-text">
          {text}
        </span>
      </label>
    );
  }
}

GroupsMenuFilterOption.defaultProps = {
  checked: false,
  disabled: false
};

GroupsMenuFilterOption.propTypes = {
  name: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
  icon: PropTypes.object.isRequired,
  setFilter: PropTypes.func.isRequired,
  checked: PropTypes.bool,
  disabled: PropTypes.bool
};

export default GroupsMenuFilterOption;
