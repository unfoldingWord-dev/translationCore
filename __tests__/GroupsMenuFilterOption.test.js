/* eslint-disable no-console */
/* eslint-env jest */
import React from 'react';
import GroupsMenuFilterOption from '../src/js/components/groupMenu/GroupsMenuFilterOption';
import { Glyphicon } from 'react-bootstrap';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('Tests for GroupsMenuFilterOption', () => {
  test('Test GroupsMenuFilterOption', () => {
    const props = {
      name: 'showMe',
      text: 'My filter option',
      icon: <Glyphicon glyph='pencil'/>,
      setFilter: jest.fn(),
      checked: true,
      disabled: false
    };

    // when
    const wrapper = shallow(
      <GroupsMenuFilterOption {...props} />
    );

    // then
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test('Test GroupsMenuFilterOption handleCheckboxChange()', () => {
    const setFilter = jest.fn();
    const props = {
      name: 'showMe',
      icon: <Glyphicon glyph='pencil'/>,
      text: 'option text',
      setFilter: setFilter,
      checked: false,
      disabled: false
    };

    // when
    const wrapper = shallow(
      <GroupsMenuFilterOption {...props} />
    );
    wrapper.find('input').simulate('change', {target: {checked: !props.checked}});

    // then
    expect(setFilter).toHaveBeenCalledWith(props.name, true);
  });

});
