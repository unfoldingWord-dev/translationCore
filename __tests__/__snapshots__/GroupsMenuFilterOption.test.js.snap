// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for GroupsMenuFilterOption Test GroupsMenuFilterOption 1`] = `
<label
  className="option"
>
  <span
    className="option-checkbox"
  >
    <input
      checked={true}
      disabled={false}
      name="showMe"
      onChange={[Function]}
      type="checkbox"
    />
  </span>
  <span
    className="option-icon"
  >
    <Glyphicon
      bsClass="glyphicon"
      glyph="pencil"
    />
  </span>
  <span
    className="option-text"
  >
    My filter option
  </span>
</label>
`;
